functions/utils/cache.test.ts

```diff
@@ -0,0 +1,30 @@
+import CacheMock from 'browser-cache-mock'
+
+import Cache from './cache'
+
+const cacheMock = new CacheMock()
+
+beforeAll(() => {
+  const globalAny: any = global
+  globalAny.caches = {
+    open: async () => cacheMock,
+    ...cacheMock,
+  }
+})
+
+test('Should use cache properly', async () => {
```

Comment:
Have a separate test for `put` and `match` so it's easy to tell which if one of them is broken.